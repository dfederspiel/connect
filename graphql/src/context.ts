import { PrismaClient } from '@prisma/client';
import { mockDeep } from 'jest-mock-extended';
import { DeepMockProxy } from 'jest-mock-extended/lib/mjs/Mock';

export type Context = {
  prisma: PrismaClient;
};

export type MockContext = {
  prisma: DeepMockProxy<PrismaClient>;
};

export const createMockContext = (): MockContext => {
  return {
    prisma: mockDeep<PrismaClient>(),
  };
};
