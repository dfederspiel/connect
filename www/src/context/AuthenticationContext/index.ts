export { useAuth, AuthProvider } from './AuthProvider';
export type { AuthContext, UserProfile } from './types';
