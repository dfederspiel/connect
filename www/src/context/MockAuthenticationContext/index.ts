export { MockAuthProvider } from './MockAuthProvider';
export { useProvideMockAuth } from './useProvideMockAuth';
