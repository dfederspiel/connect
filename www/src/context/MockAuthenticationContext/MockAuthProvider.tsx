import { useProvideMockAuth } from '.';
import { AuthProvider } from '../AuthenticationContext/';
import { AuthContext } from '../AuthenticationContext/types';
// import { AuthModule, Mode } from "./AuthModule";
/**
 * AuthContext to be passed to any component that depends on user context.
 */
// const authContext = createContext<AuthContext>(null as any);

// Provider component that wraps your app and makes auth object ...
// ... available to any child component that calls useAuth().
// eslint-disable-next-line react/prop-types

interface MockAuthProviderProps {
  origin?: string;
  children?: JSX.Element;
  user?: string;
}

export const MockAuthProvider = (props: MockAuthProviderProps): JSX.Element => {
  const { origin, children, user } = props;
  const auth = useProvideMockAuth(user) as AuthContext;
  return (
    <AuthProvider origin={origin} provider={auth}>
      {children}
    </AuthProvider>
  );
};
