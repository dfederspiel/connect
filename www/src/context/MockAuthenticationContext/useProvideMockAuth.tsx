import { AuthContext } from '../AuthenticationContext/types';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface User {}

export const useProvideMockAuth = (name?: string): AuthContext => {
  const user = name;

  const signin = async (userPolicyId?: string): Promise<void> => {
    if (userPolicyId) sessionStorage.setItem('currentPolicy', userPolicyId || '');
  };

  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const signout = (): void => {};

  const token = async (): Promise<string | null> => '123';

  return {
    user,
    signin,
    token,
    signout,
  };
};
