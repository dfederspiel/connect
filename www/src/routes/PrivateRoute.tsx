import { RouteProps } from 'react-router-dom';
import { useAuth } from '../context/AuthenticationContext';

const PrivateRoute = ({ children }: RouteProps) => {
  const auth = useAuth();
  return <>{auth.user ? children : <h1>Not Authorized</h1>}</>;
};

export default PrivateRoute;
