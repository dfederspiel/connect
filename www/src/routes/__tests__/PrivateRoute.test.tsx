import { MockAuthProvider } from '../../context/MockAuthenticationContext';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PrivateRoute from '../PrivateRoute';

describe('the private route', () => {
  it('blocks unauthenticated access', () => {
    const { asFragment } = render(
      <MockAuthProvider>
        <MemoryRouter>
          <PrivateRoute>This will NOT be in the snapshot</PrivateRoute>
        </MemoryRouter>
      </MockAuthProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('allows authenticated access', () => {
    const { asFragment } = render(
      <MockAuthProvider user="user@contoso.com">
        <MemoryRouter>
          <PrivateRoute>This will be in the snapshot</PrivateRoute>
        </MemoryRouter>
      </MockAuthProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
