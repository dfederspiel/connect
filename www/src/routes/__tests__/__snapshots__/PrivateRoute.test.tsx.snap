// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the private route allows authenticated access 1`] = `
<DocumentFragment>
  This will be in the snapshot
</DocumentFragment>
`;

exports[`the private route blocks unauthenticated access 1`] = `
<DocumentFragment>
  <h1>
    Not Authorized
  </h1>
</DocumentFragment>
`;
